model_compression_toolkit/keras/gradient_ptq/gptq_training.py

```diff
@@ -0,0 +1,144 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from typing import Callable, List, Tuple
+import tensorflow as tf
+from tqdm import tqdm
+
+from model_compression_toolkit import common
+from model_compression_toolkit.common.gptq.gptq_training import GPTQTrainer
+from model_compression_toolkit.common.gptq.gptq_config import GradientPTQConfig
+from model_compression_toolkit.common import Graph
+from model_compression_toolkit.keras.back2framework.model_builder import model_builder
+from model_compression_toolkit.common.model_builder_mode import ModelBuilderMode
+from model_compression_toolkit.keras.gradient_ptq.graph_update import update_graph_after_gptq
+from model_compression_toolkit.keras.gradient_ptq.graph_info import get_compare_points, \
+    get_trainable_parameters, get_weights_for_loss
+from model_compression_toolkit.common.framework_info import FrameworkInfo
+import numpy as np
+
+
+class KerasGPTQTrainer(GPTQTrainer):
+    """
+    Keras GPTQ training class for fine-tuning a quantized model
+    """
+    def __init__(self,
+                 graph_float: Graph,
+                 graph_quant: Graph,
+                 gptq_config: GradientPTQConfig,
+                 fw_info: FrameworkInfo):
+        """
+        Build two models from a graph: A teacher network (float model) and a student network (quantized model).
+        Use the dataset generator to pass images through the teacher and student networks to get intermediate
+        layers outputs. Use the outputs to compute the observed loss and to back-propagate the error
+        in the student network, to minimize it in the next similar steps.
+        All parameters (such as number of iterations, optimizer, etc.) are in GradientPTQConfig.
+        Args:
+            graph_float: Graph to build a float networks from.
+            graph_quant: Graph to build a quantized networks from.
+            gptq_config: GradientPTQConfig with parameters about the tuning process.
+            fw_info: Framework information
+        """
+        super().__init__(graph_float, graph_quant, gptq_config, fw_info)
+
+        # ----------------------------------------------
+        # Build two models and create compare nodes
+        # ----------------------------------------------
+        self.input_scale = 1
+        self.compare_points, _, self.compare_points_mean, self.compare_points_std = get_compare_points(
```

Comment:
Why does lines 60->71 not in common?

Reply:
We didn't decide how to do this compare point yet. currently it returns 4 outputs. I found it too strictly to be common. 
I think after we know the structure we can move share code to common.

Reply:
Let talk about it